plugins/ReactNativeOpenFile/setupBackend.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+'use strict';
+
+var getData = require('../../backend/getData');
+var getDataFiber = require('../../backend/getDataFiber');
+
+import type Bridge from '../../agent/Bridge';
+import type Agent from '../../agent/Agent';
+
+module.exports = function setupRNOpenFile(bridge: Bridge, agent: Agent, openFileInEditor: () => void) {
+  bridge.on('rn-open-file', val => {
+    // '__REACT_DEVTOOLS_GLOBAL_HOOK__.$inst'
+    var el = val.split('.').reduce(
+      (prev, curr) => prev ? prev[curr] : null,
+      window
+    );
+    var inst = el._reactInternalInstance;
```

Comment:
Sorry I wasn't clear, I didn't mean importing `getData` itself. I meant accessing the `node` data structure (that is generated by `getData` when renderer attaches). You are still accessing the "opaque nodes" (`el`) which is a problem (they're implementation details).

Agent has access to "data" in `this.elementData`. However I don't think it's meant to be a public field.

I would look at `showComponentSource` in `Panel` instead. This looks like related code (not sure if it's used though). Perhaps this is a better place to implement it, and somehow inject an implementation there?